### ISSUE TITLE
`MaxLineLength` does not exclude Markdown links followed by only punctuation

### ISSUE THREAD

**Issue by cyyynthia** [detekt/detekt#9335]
## Expected Behavior
Thanks to #5683, I'd expect the MaxLineLength rule to tolerate long lines caused by Markdown links, including when the link is followed by a punctuation mark.

## Observed Behavior
The MaxLineLength rule complains again about links if a period is added. This clashes with EndOfSentenceFormat, which expects sentences to end with a period.

## Steps to Reproduce
Lint the following snippet using Detekt:
```kt
/**
 * Annotation to prevent a function from encountering bugs, a good idea according to [my 1-bit bloom filter](https://www.xkcd.com/2934/).
 */
annotation class NeverEncounterBugs
```

Removing the period pleases the MaxLineLength rule, but triggers EndOfSentenceFormat instead. While the example is humourus and could be written on 2 lines, it gets more problematic with very long links that are longer than the permitted length on their own:

`[test](https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-controller/ann-requestmapping.html#mvc-ann-requestmapping-registration).`

## Context
I assume the intent behind #5683 is to allow that kind of long lines, and not accounting for punctuation is an oversight.

## Your Environment

* Version of detekt used: 2 alpha3
* Version of Gradle used: 9.5.1
* Operating System and version: Windows 11

**Comment by BraisGabin** [comment]
Your link to xkcd made me laugh 😂 so I decide to grab this one.
